execution/engine/emr_engine.go

```diff
@@ -172,6 +172,17 @@ func (emr *EMRExecutionEngine) Execute(executable state.Executable, run state.Ru
 }
 
 func (emr *EMRExecutionEngine) generateApplicationConf(executable state.Executable, run state.Run, manager state.Manager) []*emrcontainers.Configuration {
+	// Define custom tags
+	customTags := fmt.Sprintf("[\"flotilla_run_id:%s\", \"team:%s\", \"kube_workflow:%s\", \"kube_task_name:%s\"]",
+		run.RunID,
+		run.Labels["team"],
+		run.Labels["kube_workflow"],
+		run.Labels["kube_task_name"],
+	)
+	existingConfigString := fmt.Sprintf("[{\"spark_url\": \"http://%%host%%:4040\", \"spark_cluster_mode\": \"spark_driver_mode\", \"cluster_name\": \"%s\"}]", run.ClusterName)
+	// Add custom tags to the existing configuration string
+	updatedConfigString := strings.TrimSuffix(existingConfigString, "]") + ", \"tags\":" + customTags + "]"
```

Comment:
Kind of feels like this could be accomplished with maps or structs and then marshalled to JSON